pkg/e2e/helpers.go

```
@@ -631,3 +638,33 @@ func WaitConfigMapEvents(s *Sandbox, namespace, name string, msgs []string, time
 		return true, nil
 	})
 }
+
+// waitForBackendConfigCRDEstablish waits for backendconfig CRD to be ensured
+// by the ingress controller.
+func waitForBackendConfigCRDEstablish(crdClient *apiextensionsclient.Clientset) error {
+	var err error
+	condition := func() (bool, error) {
+		bcCRD, err := crdClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(context.TODO(), backendConfigCRDName, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				klog.V(3).Infof("CRD %s is not found, retrying", backendConfigCRDName)
+				return false, nil
+			}
+			return false, err
+		}
+		for _, c := range bcCRD.Status.Conditions {
+			if c.Type == apiextensionsv1beta1.Established && c.Status == apiextensionsv1beta1.ConditionTrue {
+				return true, nil
+			}
+		}
+		klog.V(3).Infof("CRD %s is not established, retrying", backendConfigCRDName)
+		return false, nil
+	}
+	if waitErr := wait.Poll(backendConfigEnsurePollInterval, backendConfigEnsurePollTimeout, condition); waitErr != nil {
+		if waitErr == wait.ErrWaitTimeout {
+			return fmt.Errorf("error timeout waiting for CRD established, last seen error: %v", err)
+		}
+		return fmt.Errorf("error waiting for CRD established,: %v", waitErr)
```

Do you really need to distinguish here? Why not just return the error?

---

Fixed it. This would be helpful when we wait on an error. But, it is not the case here.